rfcs/convergence/slot-null-rendering.md

```diff
@@ -0,0 +1,261 @@
+  const state = {
+    components: {
+      input: 'input',
+    },
+    input: resolveShorthand(props.input, {
+      children: React.Fragment, // 🚨 getSlots requires children
+    }),
+  };
+  const { slots, slotProps } = getSlots(state, ['input']);
+  delete slotProps.input.children; // 🚨 input can't have children
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.input {...slotProps.input} />
+    </slots.root>
+  );
+}
+```
+
+[Live Example](https://stackblitz.com/edit/react-ts-wwguyp?file=Input.tsx)
+
+### Slot as a parent for other slots
+
+This edge case has been seen in different converged components ([`CompoundButton`](https://github.com/microsoft/fluentui/blob/master/packages/react-button/src/components/CompoundButton/renderCompoundButton.tsx#L18), [`AccordionHeader`](https://github.com/microsoft/fluentui/blob/master/packages/react-accordion/src/components/AccordionHeader/renderAccordionHeader.tsx#L15))
+
+The problem is that using a slot as parent for other slots means overriding `children` props, so `children` is not declared in the moment `resolveShorthand` is invoked but in the moment that the slot will be rendered.
+
+```tsx
+export function Component(props) {
+  const state = {
+    components: {
+      button: 'button',
+      icon: 'i',
+    },
+    button: resolveShorthand(props.button, {
+      children: React.Fragment, // 🚨 getSlots requires children
+    }),
+    icon: resolveShorthand(props.icon),
+  };
+
+  const { slots, slotProps } = getSlots(state, ['input']);
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.button {...slotProps.button}>
+        <slots.icon {...slotProps.icon} />
+      </slots.button>
+    </slots.root>
+  );
+}
+```
+
+[Live Example](https://stackblitz.com/edit/react-ts-wwguyp?file=SlotAsParent.tsx)
+
+### Native elements without children
+
+e.g: An Icon that doesn't have children `<i class="icon"/>` (this cases is hypothetical, but quite possible, as we cannot verify it in converged components)
+
+## Detailed Design or Proposal
+
+Both options here proposed by the end are equivalent, the only difference is the impact on `ObjectShorthandProps` interface
+
+### Option 1: Make `shorthands` optional
+
+This solves this problem by verifying if the shorthand is `undefined` or not.
+By verifying if shorthand is `undefined` we can opt for null rendering without compromising the cases which native slots don't have children
+
+- `getSlots` should stop verifying null rendering by the existence of the property `children`, instead depending on the existence of the shorthand itself
+
+```tsx
+// react-utilities/src/compose/getSlots.ts
+function getSlots(state, slotNames) {
+  // ...
+  if (state[name] === undefined) {
+    slots[name] = nullRender;
+    continue;
+  }
+  // ...
+}
+```
+
+- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even presenting default props
```

Comment:
nit:
```suggestion
- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even when presented with default props.
```